docs/extensions/storage/_index.md

```diff
@@ -8,39 +8,144 @@ geekdocFilePath: _index.md
+
+The dashed lines in the diagram indicate requests that are made to authenticate requests or lookup the storage provider:
+1. After authenticating a request, the proxy may either use the CS3 `userprovider` or the accounts service to fetch the user information that will be minted into the `x-access-token`.
+2. The gateway will verify the JWT signature of the `x-access-token` or try to authenticate the request itself, e.g. using a public link token.
+
+{{< hint warning >}}
+The bottom part is lighter because we will deprecate it in favor of using only the CS3 user and group providers after moving some account functionality into reva and glauth. The metadata storage is not registered in the reva gateway to  seperate metadata necessary for running the service from data that is being served directly.
+{{< /hint >}}
+
+## Terminology
+
+Communication is hard. And clear communication is even harder. You may encounter the following terms throughout the documentation, in the code or when talking to other developers. Just keep in mind that whenever you hear or read *storage*, that term needs to be clarified, because on its own it is too vague. PR welcome.
+
+### Resources
+A *resource* is a logical concept. Ressources can be of [different types](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.ResourceType):
+- an actual *file*
+- a *container*, e.g. a folder or bucket
+- a *symlink*, or
+- a *reference* which can point to a resource in another *storage provider*
+
+### References
+
+A [*reference*](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.Reference) is a logical concept. It identifies a *resource* and consists of either
+- a *path* based reference, used to identify a *resource* in the *namespace* of a *storage provider*. It must start with a `/`.
+- an [*id* based reference](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.ResourceId), uniquely identifying a *resource* in the *namespace* of a *storage provider*. It consists of a `storage provider id` and an `opaque id`. The `storage provider id` must NOT start with a `/`.
+
+{{< hint info >}}
+The `/` is important because currenty the static *storage registry* uses a map to look up which *storage provider* is responsible for the resource. Paths must be prefixed with `/` so there can be no collisions between paths and storage provider ids in the same map.
+{{< /hint >}}
+
+{{< hint warning >}}
+**Proposed Change**
+
+A *reference* is a logical concept. It identifies a *resource* and consists of
+a *root* and a *path* relative to that *root*. 
+A *root* is a `storage space id` and a `logical uuid` of a resource inside the *storage space*:
+```
+<storages_pace>:<logical_uuid>:<relative_path>
+`------------root------------´ `----path-----´
+```
+*root* and *path* are both optional. The `storage space id` should be a uuid to prevent references from breaking when a *user* or *storage space* gets renamed. The following are all valid references:
+
```

Comment:
I think what you have in mind is closer to the currently used types of references in CS3:
- an ID based reference, consisting of `storageid` and `opaqueid`
- a path based reference, consisting of a `path`

Most requests come with path based references in the global namespace. The global namespace could use aliases for storage spaces instead of ids. The gateway will look up which storage provider is responsible for the path in the storage registry, adjust the path based reference by ltrimming the mount path ond the storeg provider will receive a path that is relative to its root.

For Id based references the gateway will also look up which storage provider is responsible in the storage registry. This time, the storage registry extracts the storageid (which identifies the storage space, not the storage provider) and either finds it in a manually configured map of storageid->addr:port or in a list of cached storage ids (with a fallback to ping all existing storage providers if they know the opaqueid / id based reference to build the cache). In this case either the gateway or the storage registry may have to replace the opaqueid with the path by calling the getPithByID method on the storage provider.